beacon_node/eth2_libp2p/src/peer_manager/peerdb.rs

```diff
@@ -357,7 +360,8 @@ impl<TSpec: EthSpec> PeerDB<TSpec> {
         let log = &self.log;
         self.peers.iter_mut()
             .filter(move |(_, info)| {
-                info.is_connected() && info.on_subnet(subnet_id)
+                // TODO(pawan): should log a message or have a metric for when metadata and gossipsub diverge
```

Comment:
Can we action or remove this TODO

Reply:
I think we should log something

Reply:
Sorry forgot about this. Will add a log.

Reply:
This is actually a bit tricky. gossipsub and metadata attnets can give conflicting results because it takes `PING_INTERVAL` (30secs) in the worst case for a peer to communicate an updated metadata.

For e.g. peer sends a ping -> peer subscribes to some subnet -> updates its metadata attnets -> wait 30 secs -> send ping with updated metadata
So we have this 30 sec period when metadata subscriptions don't agree with gossipsub subscriptions.

Reply:
So I think we should drop the logging here as it won't give us useful info most of the times.